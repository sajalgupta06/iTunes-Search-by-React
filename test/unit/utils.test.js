import {
  capitalize,
  getApiUrl,
  getMedia,
  getKind
} from '../../src/utils';

describe('utils.js', () => {
  test('capitalize should be able to capitalize words', () => {
    const result = 'Abc';
    expect(capitalize('abc')).toMatchSnapshot();
    expect(capitalize('aBc')).toMatchSnapshot();
    expect(capitalize('ABC')).toMatchSnapshot();
  });

  test('getMedia should return proper term for API usage', () => {
    expect(getMedia('TV Show')).toMatchSnapshot();
  });

  test('getApiUrl should return correct API endpoint', () => {
    expect(getApiUrl({
      media: 'music',
      query: 'bsb'
    })).toMatchSnapshot();
  });

  test('getApiUrl should return correct API endpoint without media type', () => {
    expect(getApiUrl({
      query: 'bsb'
    })).toMatchSnapshot();
  });

  test('getKind should return correct format for UI', () => {
    expect(getKind('feature-movie')).toMatchSnapshot();
    expect(getKind('tv')).toMatchSnapshot();
    expect(getKind('song')).toMatchSnapshot();
  });
});
