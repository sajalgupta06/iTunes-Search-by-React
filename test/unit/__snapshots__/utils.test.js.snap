// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`utils.js capitalize should be able to capitalize words 1`] = `"Abc"`;

exports[`utils.js capitalize should be able to capitalize words 2`] = `"Abc"`;

exports[`utils.js capitalize should be able to capitalize words 3`] = `"Abc"`;

exports[`utils.js getApiUrl should return correct API endpoint 1`] = `"https://itunes.apple.com/search?media=music&term=bsb"`;

exports[`utils.js getApiUrl should return correct API endpoint without media type 1`] = `"https://itunes.apple.com/search?media=all&term=bsb"`;

exports[`utils.js getKind should return correct format for UI 1`] = `"Movie"`;

exports[`utils.js getKind should return correct format for UI 2`] = `"TV"`;

exports[`utils.js getKind should return correct format for UI 3`] = `"Song"`;

exports[`utils.js getMedia should return proper term for API usage 1`] = `"tvShow"`;
